app/controllers/concerns/authorized.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+module Authorized
+  def admin
+    render json: { errors: 'Unauthorized access' }, status: :forbidden unless admin?
+  end
+
+  def admin?
+    @current_user.role.name == 'admin'
+  end
+
+  def owner?
+    params[:id].to_i == @current_user.id
+  end
+
+  def ownership?
+    render json: { errors: 'Unauthorized access' }, status: :forbidden unless admin? || owner?
+  end
+end
```

Comment:
No olvides configurar el EOF en tu editor de código para evitar este error:
```suggestion
end

```